backend/internal/controllers/router.go

```diff
@@ -42,25 +45,31 @@ func GetRouter(authenticator auth.Authenticator) chi.Router {
 	r.Get("/ping", ping)
 
 	r.Route("/books", func(r chi.Router) {
-		// "localhost:8080/books/
 		r.Get("/", bookController.GetBookList)
 
-		// "localhost:8080/books/{id}/{lang}
 		r.Get("/{id}/{lang}", bookController.GetBookDetails)
 
-		r.Post("/", bookController.CreateBook)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanUploadBooks)).Post("/", bookController.CreateBook)
 
-		r.Post("/{id}", bookController.CreateBookDetail)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanUploadBooks)).Post("/{id}", bookController.CreateBookDetail)
 
-		r.Delete("/{id}", bookController.DeleteBook)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanDeleteBooks)).Delete("/{id}", bookController.DeleteBook)
 
-		r.Delete("/{id}/{lang}", bookController.DeleteBookDetail)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanDeleteBooks)).Delete("/{id}/{lang}", bookController.DeleteBookDetail)
 
-		r.Patch("/{id}", bookController.UpdateBook)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanEditBooks)).Patch("/{id}", bookController.UpdateBook)
 
-		r.Patch("/{id}/{lang}", bookController.UpdateBookDetails)
+		r.With(middleware.RequireAuth(authenticator)).
+			With(middleware.RequirePermission(adminDB, models.CanEditBooks)).Patch("/{id}/{lang}", bookController.UpdateBookDetails)
```

Comment:
Small thing, `r.With()` is variadic so you can clean this up by combining the RequireAuth and RequirePermission middlewares.